presto-main/src/main/java/com/facebook/presto/metadata/FunctionManager.java

```diff
@@ -1,5 +1,199 @@
+    }
+
+    public boolean canResolveOperator(Session session, OperatorType operatorType, Type returnType, List<? extends Type> argumentTypes)
+    {
+        for (SqlPathElement element : session.getPath().getParsedPath()) {
+            FunctionNamespace namespace = functionNamespaces.get(fetchCatalog(element, session));
+            if (namespace != null && namespace.canResolveOperator(operatorType, returnType, argumentTypes)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean isRegistered(FunctionHandle handle)
+    {
+        return functionNamespaces.get(handle.getCatalog()).isRegistered(handle.getSignature());
+    }
+
+    public FunctionHandle resolveOperator(Session session, OperatorType operatorType, List<? extends Type> argumentTypes)
+            throws OperatorNotFoundException
+    {
+        for (SqlPathElement element : session.getPath().getParsedPath()) {
+            FunctionNamespace namespace = functionNamespaces.get(fetchCatalog(element, session));
+            if (namespace != null) {
+                try {
+                    Signature signature = namespace.resolveOperator(operatorType, argumentTypes);
+                    return new FunctionHandle(signature, fetchCatalog(element, session), TEMP_DEFAULT_SCHEMA);
+                }
+                catch (PrestoException e) {
+                    if (!(e instanceof OperatorNotFoundException)) {
+                        throw e;
+                    }
+                }
+            }
+        }
+        throw new OperatorNotFoundException(
+                operatorType,
+                argumentTypes.stream()
+                        .map(Type::getTypeSignature)
+                        .collect(toImmutableList()));
+    }
+
+    public FunctionHandle getCoercion(Session session, Type fromType, Type toType)
+    {
+        return getCoercion(session, fromType.getTypeSignature(), toType.getTypeSignature());
+    }
+
+    public FunctionHandle getCoercion(Session session, TypeSignature fromType, TypeSignature toType)
+    {
+        for (SqlPathElement element : session.getPath().getParsedPath()) {
+            FunctionNamespace namespace = functionNamespaces.get(fetchCatalog(element, session));
+            if (namespace != null) {
+                try {
+                    Signature signature = namespace.getCoercion(fromType, toType);
+                    return new FunctionHandle(signature, fetchCatalog(element, session), TEMP_DEFAULT_SCHEMA);
+                }
+                catch (PrestoException e) {
+                    if (!(e instanceof OperatorNotFoundException)) {
+                        throw e;
+                    }
+                }
+            }
+        }
+        throw new OperatorNotFoundException(OperatorType.CAST, ImmutableList.of(fromType), toType);
+    }
+
+    private String fetchCatalog(SqlPathElement pathElement, Session session)
```

Comment:
`fetchCatalog` -> `getCatalog`. ("Fetch" sounds like we are doing a remote call to fetch it).